timelock-agent/src/main/java/com/palantir/atlasdb/timelock/paxos/SingleLeaderNetworkClientFactories.java

```diff
@@ -75,17 +78,30 @@
     @Override
     public Factory<PaxosLearnerNetworkClient> learner() {
         return client -> {
-            List<PaxosLearner> remoteLearners = TimelockPaxosLearnerAdapters
-                    .create(useCase(), remoteClients(), useBatchedEndpoints(), client);
+            List<WithDedicatedExecutor<PaxosLearner>> remoteLearners = TimelockPaxosLearnerAdapters
+                    .create(useCase(), remoteClients(), useBatchedEndpoints(), client)
+                    .stream()
+                    .map(withExecutor -> withExecutor.transformService(
+                            remote -> metrics().instrument(PaxosLearner.class, remote, client)))
+                    .collect(Collectors.toList());
             PaxosLearner localLearner = components().learner(client);
 
-            LocalAndRemotes<PaxosLearner> allLearners = LocalAndRemotes.of(localLearner, remoteLearners)
-                    .enhanceRemotes(remote -> metrics().instrument(PaxosLearner.class, remote, client));
+            LocalAndRemotes<PaxosLearner> allLearners = LocalAndRemotes.of(localLearner,
+                    remoteLearners.stream().map(WithDedicatedExecutor::service).collect(Collectors.toList()));
```

Comment:
This doesn't feel necessary - is there a reason why you can't just pass in local and remote (mapping away the `WithExecutor`) into the constructor at the end?

Reply:
Yep, no reason we couldn't - I guess this must have been more used previously. I've removed it.